scipy/optimize/_lsq/givens_elimination.pyx

```diff
@@ -44,7 +47,7 @@ def givens_elimination(double [:, ::1] S, double [:] v, double [:] diag):
         if diag[i] == 0:
             continue
 
-        diag_row[:] = 0
+        diag_row[i+1:] = 0
```

Comment:
It's not obvious to me that this change is correct.

The code currently starts by setting ``diag_row`` to all zeros except for element ``i``. Then in the inner loop it touches only elements ``i..n``. The comment ``diag_row[j] is implicitly 0 now`` is not very clear. The new and old code are only equivalent is ``diag_row[i]`` really is reset to 0 within ``dlartg`` it looks like to me.

@nmayorov you want to have a look at this one?